lib/puppet/pops/validation/checker4_0.rb

```diff
@@ -213,11 +213,24 @@ def check_BinaryExpression(o)
     rvalue(o.right_expr)
   end
 
+  def resource_without_title?(o)
+    if o.instance_of?(Puppet::Pops::Model::BlockExpression)
+      statements = o.statements
+      statements.length == 2 && statements[0].instance_of?(Puppet::Pops::Model::QualifiedName) && statements[1].instance_of?(Puppet::Pops::Model::LiteralHash)
+    else
+      false
+    end
+  end
+
   def check_BlockExpression(o)
-    o.statements[0..-2].each do |statement|
-      if idem(statement)
-        acceptor.accept(Issues::IDEM_EXPRESSION_NOT_LAST, statement)
-        break # only flag the first
+    if resource_without_title?(o)
+      acceptor.accept(Issues::RESOURCE_WITHOUT_TITLE, o, :name => o.statements[0].value)
+    else
+      o.statements[0..-2].each do |statement|
+        if idem(statement)
+          acceptor.accept(Issues::IDEM_EXPRESSION_NOT_LAST, statement)
+          break # only flag the first
+        end
```

Comment:
I can answer my own question by looking at the additional tests; apparently there is always a block produced of size 2 (just what the predicate method checks) instead of a some kind of broken resource expression, or keeping the two expressions in questions in the same list of statements as what surrounds it. This fix is good enough for now as it is covered by tests - at some point the grammar may be refactored in this area and the tests will then catch that.